collection-scripts/gather

```diff
@@ -1,40 +1,24 @@
 #!/bin/bash
+#
+# This is the main script that is invoked when we call `oc adm must-gather`,
+# and it is called directly regardless of the ENTRYPOINT in the Dockerfile
+#
+# This file is necessary to pipe the output of the must-gather process to the
+# output directory.
 
-DIR_NAME=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-# shellcheck disable=SC1091
-source "${DIR_NAME}/common.sh"
+LOG_DIR="${BASE_COLLECTION_PATH:-/must-gather}"
 
-# get apiservices
-/usr/bin/gather_apiservices
+# If directory doesn't exists it's because we are manually creating the
+# container for development, so we'll use a temporary directory
+if [[ ! -d "$LOG_DIR" ]]; then
+  LOG_DIR=`mktemp -d`
+  export BASE_COLLECTION_PATH="$LOG_DIR"
+  OC_CALLED=false
```

Comment:
one last question before merging this change.
what's the purpose of `OC_CALLED` ? I think I understand the meaning from the name but I'm not sure I got how it's used in the whole flow and which condition uses it (but it's probably my bad here

Reply:
This is leftover stuff I had from a previous version of the patch, in that one when OC_CALLED was false it would enter an endless loop.

I'll remove it now, thanks for catching this.